src/server/mailgun.ts

```diff
@@ -43,10 +58,201 @@ export const sendWelcomeEmail = async (emailTo: string, fullName: string) => {
+                    studyURL: `${BASE_URL}/organization/${org?.slug}/study/${studyId}/review`,
+                }),
+            })
+        } catch (error) {
+            logger.error.log('sendStudyProposalEmails error: ', error)
+        }
+    }
+}
+
+export const sendStudyProposalApprovedEmail = async (studyId: string) => {
+    const [mg, domain] = await mailGunConfig()
+    if (!mg) return
+
+    const { study, org } = await getStudyAndOrg(studyId)
+
+    const researchersToNotify = await getUsersByRoleAndOrgId('researcher', org.id)
+
+    for (const researcher of researchersToNotify) {
+        const email = researcher.email
+        if (!email) continue
+
+        try {
+            await mg.messages.create(domain, {
+                from: SI_EMAIL,
+                to: email,
+                subject: 'SafeInsights - Study Proposal Approved',
+                template: 'vb - research proposal approved',
+                'h:X-Mailgun-Variables': JSON.stringify({
+                    fullName: researcher.fullName,
+                    studyTitle: study.title,
+                    submittedBy: study.researcherName,
+                    submittedTo: org.name,
+                    submittedOn: dayjs(study.createdAt).format('MM/DD/YYYY'),
+                }),
+            })
+        } catch (error) {
+            logger.error.log('sendStudyProposalApprovedEmail error: ', error)
+        }
+    }
+}
+
+export const sendStudyProposalRejectedEmail = async (studyId: string) => {
+    const [mg, domain] = await mailGunConfig()
+    if (!mg) return
+
+    const { study, org } = await getStudyAndOrg(studyId)
+
+    const researchersToNotify = await getUsersByRoleAndOrgId('researcher', org.id)
+
+    for (const researcher of researchersToNotify) {
+        const email = researcher.email
+        if (!email) continue
+
+        try {
+            await mg.messages.create(domain, {
+                from: SI_EMAIL,
+                to: email,
+                subject: 'SafeInsights - Study Proposal Rejected',
+                template: 'vb - research proposal rejected',
+                'h:X-Mailgun-Variables': JSON.stringify({
+                    fullName: researcher.fullName,
+                    studyTitle: study.title,
+                    submittedBy: study.researcherName,
+                    submittedTo: org.name,
+                    submittedOn: dayjs(study.createdAt).format('MM/DD/YYYY'),
+                    studyURL: `${BASE_URL}/organization/${org.slug}/study/${studyId}/review`,
+                }),
+            })
+        } catch (error) {
+            logger.error.log('sendStudyProposalRejectedEmail error: ', error)
+        }
+    }
+}
+
+export const sendResultsReadyForReviewEmail = async (studyId: string) => {
+    const [mg, domain] = await mailGunConfig()
+    if (!mg) return
+
+    const { study, org } = await getStudyAndOrg(studyId)
+    const reviewersToNotify = await getUsersByRoleAndOrgId('reviewer', org.id)
```

Comment:
for this email we're only supposed to notify the reviewer who initially reviewed the study